service/history/execution/mutable_state_builder_methods_signal_test.go

```diff
@@ -0,0 +1,181 @@
+package execution
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/types"
+	"github.com/uber/cadence/service/history/constants"
+	"testing"
+)
+
+func Test__IsSignalRequested(t *testing.T) {
+	mb := testMutableStateBuilder(t)
+	requestID := "101"
+	t.Run("signal not found", func(t *testing.T) {
+		result := mb.IsSignalRequested(requestID)
+		assert.False(t, result)
+	})
+	t.Run("signal found", func(t *testing.T) {
+		mb.pendingSignalRequestedIDs[requestID] = struct{}{}
+		result := mb.IsSignalRequested(requestID)
+		assert.True(t, result)
+	})
+}
+
+func Test__GetSignalInfo(t *testing.T) {
+	mb := testMutableStateBuilder(t)
+	initiatedEventID := int64(1)
+	info := &persistence.SignalInfo{
+		InitiatedID:     1,
+		SignalRequestID: "101",
+	}
+	t.Run("signal not found", func(t *testing.T) {
+		_, ok := mb.GetSignalInfo(initiatedEventID)
+		assert.False(t, ok)
+	})
+	t.Run("signal found", func(t *testing.T) {
+		mb.pendingSignalInfoIDs[initiatedEventID] = info
+		result, ok := mb.GetSignalInfo(initiatedEventID)
+		assert.True(t, ok)
+		assert.Equal(t, info, result)
+	})
+}
+
+func Test__ReplicateExternalWorkflowExecutionSignaled(t *testing.T) {
+	mb := testMutableStateBuilder(t)
+	event := &types.HistoryEvent{
+		ExternalWorkflowExecutionSignaledEventAttributes: &types.ExternalWorkflowExecutionSignaledEventAttributes{
+			InitiatedEventID: 1,
+		},
+	}
+	info := &persistence.SignalInfo{
+		InitiatedID:     1,
+		SignalRequestID: "101",
+	}
+	mb.pendingSignalInfoIDs[int64(1)] = info
+	mb.updateSignalInfos[int64(1)] = info
+	err := mb.ReplicateExternalWorkflowExecutionSignaled(event)
+	assert.NoError(t, err)
+	assert.NotNil(t, mb.deleteSignalInfos[int64(1)])
+}
+
+func Test__ReplicateSignalExternalWorkflowExecutionFailedEvent(t *testing.T) {
+	mb := testMutableStateBuilder(t)
+	event := &types.HistoryEvent{
+		SignalExternalWorkflowExecutionFailedEventAttributes: &types.SignalExternalWorkflowExecutionFailedEventAttributes{
+			InitiatedEventID: 1,
+		},
+	}
+	info := &persistence.SignalInfo{
+		InitiatedID:     1,
+		SignalRequestID: "101",
+	}
+	mb.pendingSignalInfoIDs[int64(1)] = info
+	mb.updateSignalInfos[int64(1)] = info
+	err := mb.ReplicateSignalExternalWorkflowExecutionFailedEvent(event)
+	assert.NoError(t, err)
+	assert.NotNil(t, mb.deleteSignalInfos[int64(1)])
```

Comment:
Not sure if possible, but this test is identical to the one above (except for the function called `ReplicateSignalExternalWorkflowExecutionFailedEvent`). Would it be possible to assert some more things, maybe? Reading the tests, it looks like the functions have identical effects?